napari/_qt/_tests/test_qt_notifications.py

```diff
@@ -64,6 +68,46 @@ def store_widget(self, *args, **kwargs):
     monkeypatch.setattr(_QtMainWindow, "current", mock_current_main_window)
 
 
+@dataclass
+class ShowStatus:
+    show_notification_count: int = 0
+    show_traceback_count: int = 0
+
+
+@pytest.fixture
+def count_show(monkeypatch, qtbot):
+
+    stat = ShowStatus()
+
+    def mock_show_notif(_):
+        stat.show_notification_count += 1
+
+    def mock_show_traceback(_):
+        stat.show_traceback_count += 1
+
+    monkeypatch.setattr(NapariQtNotification, "show", mock_show_notif)
+    monkeypatch.setattr(TracebackDialog, "show", mock_show_traceback)
+
+    return stat
+
+
+@pytest.fixture(autouse=True)
+def ensure_qtbot(monkeypatch, qtbot):
+    old_notif_init = NapariQtNotification.__init__
+    old_traceback_init = TracebackDialog.__init__
+
+    def mock_notif_init(self, *args, **kwargs):
+        old_notif_init(self, *args, **kwargs)
+        qtbot.add_widget(self)
+
+    def mock_traceback_init(self, *args, **kwargs):
+        old_traceback_init(self, *args, **kwargs)
+        qtbot.add_widget(self)
+
+    monkeypatch.setattr(NapariQtNotification, "__init__", mock_notif_init)
```

Comment:
Will the class definition (e.g. `NapariQtNotification`) be imported fresh for each test function? Or is there a risk that `old_notif_init` may be the definition from the last test function run?

Reply:
pytest monkeypatch, restore previous value on test end